BUILDING.md

```diff
@@ -0,0 +1,79 @@
+
+# Building CircuitPython
+
+Welcome to CircuitPython!  
+
+This document is a quick-start guide only.
+
+Detailed guides on how to build CircuitPython can be found in the Adafruit Learn system at
+https://learn.adafruit.com/building-circuitpython/
+
+## Setup
+
+Please ensure you setup your build environment appropriately, as per the guide.  You will need:
+
+* Linux: https://learn.adafruit.com/building-circuitpython/linux
+* MacOS: https://learn.adafruit.com/building-circuitpython/macos
+* Windows Subsystem for Linux (WSL): https://learn.adafruit.com/building-circuitpython/windows-subsystem-for-linux
+
+### Submodules
+
+This project has a bunch of git submodules.  You will need to update them regularly.
+
+    git submodule sync
+    git submodule update --init --recursive
```

Comment:
I'd rather not do recursive here because it overrides .gitmodule settings. Instead we should be able to configure them correctly there.

Reply:
Hmm, I have always done — recursive
What should I be doing?

Reply:
@tannewt jerry raises a good point - i've only ever done --recursive!  What's the right thing to do?

Reply:
I believe dropping `--recursive` from the command should work.

Dropping it allows us to ignore specific recursive submodules like the ones TinyUSB has for its example code: https://github.com/adafruit/circuitpython/blob/master/.gitmodules#L80

I believe `--recursive` overrides this setting according to this stack overflow: https://stackoverflow.com/questions/58403616/does-git-submodule-update-recursive-override-fetchrecursesubmodules

Reply:
OK removed.

Reply:
so just ```git submodule update --init``` is sufficient or is there some other incantation we need?

Reply:
@jerryneedell I believe so. `git submodule sync` is also need as before.

Reply:
> @jerryneedell I believe so. `git submodule sync` is also need as before.

OK -- seems to work without --recursive.  I tried it with --recursive and it broke -- so I've stopped using --recursive